include/odp/api/spec/packet.h

```diff
@@ -235,6 +246,18 @@ int odp_packet_reset(odp_packet_t pkt, uint32_t len);
  */
 odp_packet_t odp_packet_from_event(odp_event_t ev);
 
+/**
+ * Convert multiple packet events to packet handles
+ *
+ * All events must be of type ODP_EVENT_PACKET.
+ *
+ * @param[out] pkt  Packet handle array for output
+ * @param      ev   Array of event handles to convert
+ * @param      num  Number of packets and events
+ */
+void odp_packet_from_event_multi(odp_packet_t pkt[], const odp_event_t ev[],
+				 int num);
```

Comment:
Even with the `odp_event_type_multi()` API semantics, it's still not clear why this is needed. Given that `odp_packet_from_event()` is likely just a cast there doesn't seem to be a great deal to be gained by having a `multi` version of this which can't be easily inlined away.

Reply:
pkts = odp_schedule_multi(NULL, ODP_SCHED_NO_WAIT, ev_tbl, MAX_PKT_BURST);

if (pkts <= 0)
	continue;

for (i = 0; i < pkts; i++)
	pkt_tbl[i] = odp_packet_from_event(ev_tbl[i]);

VS.

odp_packet_from_event_multi(pkt_tlb, ev_tbl, num_pkts);

1) Single call vs. many calls makes difference when not inlined (ABI compat).
2) This is not only cast, it's also a copy of the handle. Implementation may be vectorized, for loop unrolled, etc.
3) Depending on implementation the conversion may include additional things to the handle copy: extra checks, post processing of the event, etc HW/implementation specific tricks. Which would open more optimization opportunity in addition to load+store.
4) Application code is cleaner